packages/integrations/useProgressBar/index.ts

```diff
@@ -0,0 +1,28 @@
+import progress, { NProgressOptions } from 'nprogress'
+import { MaybeRef, tryOnUnmounted } from '@vueuse/shared'
+import { ref, watch, isRef } from 'vue-demi'
+
+/**
+ * Reactive progress bar.
+ *
+ * @see   {@link https://vueuse.js.org/useProgressBar}
+ * @param newIcon
+ * @param options
+ */
+export function useProgressBar(
+  current: MaybeRef<boolean | null | undefined> = null,
+  options?: NProgressOptions | undefined,
+) {
+  const loading = isRef(current)
+    ? current
+    : ref<boolean | null>(current)
+
+  if (options)
+    progress.configure(options)
+
+  watch(loading, active => active ? progress.start() : progress.done())
+
+  tryOnUnmounted(progress.remove)
+
+  return loading
```

Comment:
Maybe it's better to return an object giving more controls to users.

```ts
{
  progress: Ref<number>,
  isLoading: Ref<boolean>,
  remove: Fn,
  done: Fn,
  /* ... */
}
```

Reply:
I think we have a problem, when I use the library functions to change the status I don't have any events that can notify me to update the refs